### PR TITLE
feat(liteLLM): Implemented image support and corresponding tests

### LINKED ISSUE #712: [instrumentation-python] litellm support for images

### PR DESCRIPTION
**Pull request by shreyabsridhar**
Resolves #712 
- Implemented image support so that input images render as in below screenshot
  - Implemented corresponding test: `test_completion_image_support()`
- Modified existing completion logic for multiple messages to align with other instrumentations
- Added 2 lines of code to support base64_json images in image generation in additional to http urls
  - Implemented corresponding test: `test_image_generation_b64json`
Screenshot 2024-08-15 at 10 58 25 AM

### CODE REVIEW

File: python/instrumentation/openinference-instrumentation-litellm/src/openinference/instrumentation/litellm/__init__.py [hunk 1]
```diff
@@ -51,10 +105,16 @@ def _instrument_func_type_completion(span: trace_api.Span, kwargs: Dict[str, Any
...
     if messages := kwargs.get("messages"):
-        _set_span_attribute(span, SpanAttributes.INPUT_VALUE, str(messages[0].get("content")))
-        for i, obj in enumerate(messages):
-            for key, value in obj.items():
-                _set_span_attribute(span, f"input.messages.{i}.{key}", value)
+        if content := messages[0].get("content"):
+            if isinstance(content, str):
+                _set_span_attribute(span, SpanAttributes.INPUT_VALUE, str(content))
+            elif is_iterable_of(content, dict):
+                _set_span_attribute(span, SpanAttributes.INPUT_VALUE, str(messages))
```
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-litellm/src/openinference/instrumentation/litellm/__init__.py:112]: I don't understand this, if the content is a dict, we set all the messages as the input value?

File: python/instrumentation/openinference-instrumentation-litellm/src/openinference/instrumentation/litellm/__init__.py [hunk 2]
```diff
@@ -112,10 +172,12 @@ def _finalize_span(span: trace_api.Span, result: Any) -> None:
...
         if len(result.data) > 0:
             if img_data := result.data[0]:
-                if isinstance(img_data, Image) and (url := img_data.url):
+                if isinstance(img_data, Image) and (url := (img_data.url or img_data.b64_json)):
                     _set_span_attribute(span, ImageAttributes.IMAGE_URL, url)
                     _set_span_attribute(span, SpanAttributes.OUTPUT_VALUE, url)
-                elif isinstance(img_data, dict) and (url := img_data.get("url")):
+                elif isinstance(img_data, dict) and (
+                    url := (img_data.get("url") or img_data.get("b64_json"))
+                ):
```
Comment (fjcasti1) [inline python/instrumentation/openinference-instrumentation-litellm/src/openinference/instrumentation/litellm/__init__.py:180]: noice

**Review by fjcasti1 — COMMENTED** [review]
lgtm but have a quick question we can sync on

File: python/instrumentation/openinference-instrumentation-litellm/src/openinference/instrumentation/litellm/__init__.py (hunk 1 above)
Comment (shreyabsridhar) [reply python/instrumentation/openinference-instrumentation-litellm/src/openinference/instrumentation/litellm/__init__.py]: Resolved with @RogerHYang and @fjcasti1- for now just setting input value to be `json.dumps(messages)`

**Review by fjcasti1 — APPROVED** [review]

**Pull request merged**